app/src/main/java/com/ataulm/whatsnext/api/LetterboxdApi.kt

```diff
@@ -26,14 +20,52 @@ interface LetterboxdApi {
     suspend fun search(@Query("input") searchTerm: String): ApiSearchResponse
 
     @GET("film/{id}")
-    fun film(@Path("id") letterboxdId: String): Single<ApiFilm>
+    suspend fun film(@Path("id") letterboxdId: String): ApiFilm
 
     @RequiresAuthenticatedUser
     @GET("film/{id}/me")
-    fun filmRelationship(@Path("id") letterboxdId: String): Single<ApiFilmRelationship>
+    suspend fun filmRelationship(@Path("id") letterboxdId: String): ApiFilmRelationship
+
+    @RequiresAuthenticatedUser
+    @PATCH("film/{id}/me")
+    suspend fun updateFilmRelationship(
+            @Path("id") letterboxdId: String,
+            @Body request: ApiFilmRelationshipUpdateRequest
+    ): ApiFilmRelationshipUpdateResponse
 }
 
 data class AuthTokenApiResponse(
         @SerializedName("access_token") val accessToken: String,
         @SerializedName("refresh_token") val refreshToken: String
 )
+
+data class ApiFilmRelationshipUpdateRequest(
+        @SerializedName("watched") val watched: Boolean,
+        @SerializedName("liked") val liked: Boolean,
+        @SerializedName("inWatchlist") val inWatchlist: Boolean,
+        /**
+         * Accepts values between 0.5 and 5.0, with increments of 0.5, or null (to remove the
+         * rating). If set, [watched] is assumed to be true.
+         */
+        @SerializedName("rating") val rating: String
+)
+
+data class ApiFilmRelationshipUpdateResponse(
+        @SerializedName("data") val data: ApiFilmRelationship,
+        @SerializedName("messages") val messages: List<ApiFilmRelationshipUpdateMessage>
+)
+
+data class ApiFilmRelationshipUpdateMessage(
```

Comment:
we don't really use this because we don't really handle errors 🤔 If you pay me for Android work, dw I'll handle errors.